### PR TITLE
A bunch of stuff

### PR DESCRIPTION
**Pull request by SpookyKipper**
1. I beautified the code a bit using a tool but it should still work

2. Added `showRedirect($domain)`
3. Moved `tableToArray($html)` out
4. Improved `getUserStats($option)`
   - The ability to get individual statistics item
   - Remove the random numebrs/elements that VistaPanel has for some reason

### CODE REVIEW

**Review by greenreader9 — COMMENTED** [review]
Honestly, looks good to me. As long as it works!

**Review by 4yx — COMMENTED** [review]
I can't say I disapprove, especially with the simpleCurl beautifications, and if it works well.

I am curious, however, about some oddities:
In line 192, the array is converted to ``["submit" => true]``. However, in line 283, it goes and does:
```php
[
                "db" => $dbname,
]
```
Instead of an one-line ``["db" => $dbname]``.

There are also some places where I'm not sure beautificiation was needed/may add unnecessary lines (like the login function parameters, or the cookie setting in login, or lines 566-571), but if you intend to use this tool from now on, changing them manually would be pointless.

**Comment by SpookyKipper** [comment]
> ```
> [
>                 "db" => $dbname,
> ]
> ```
> 
> Instead of an one-line `["db" => $dbname]`.

It seems that it's because it follows if the previous `array(variable)` is a single line or multi-line.


> There are also some places where I'm not sure beautificiation was needed/may add unnecessary lines (like the login function parameters, or the cookie setting in login, or lines 566-571), but if you intend to use this tool from now on, changing them manually would be pointless.

Perhaps it got confused because we skip lines for like 1 or 2 strings and not having a specific pattern

**Review by 4yx — APPROVED** [review]
Hmm, okay. See if it's easy to fix those (and if you believe they should be fixed, of course), otherwise everything looks good to me as well.

**Comment by SpookyKipper** [comment]
I actually don't mind if it's fixed or not, since it won't make a big difference.

**Comment by ImLoadingUuU** [comment]
Cool. i know random numbers problem in vista panel. but idk how to fix that.

**Comment by 4yx** [comment]
> I actually don't mind if it's fixed or not, since it won't make a big difference.

Planning to merge it then?

File: api-client.php [hunk 1]
```diff
@@ -198,94 +241,123 @@ public function login($username = "", $password = "", $theme = "PaperLantern")
...
         $this->checkLogin();
         $this->checkForEmptyParams($dbname);
-        $this->simpleCurl($this->cpanelUrl . "/panel/indexpl.php?option=mysql&cmd=create", true, array(
-            "db" => $dbname
-        ), false, array(
-            $this->cookie
-        ));
+        $this->simpleCurl(
+            $this->cpanelUrl . "/panel/indexpl.php?option=mysql&cmd=create",
+            true,
```
Comment (greenreader9) [inline api-client.php:282]: Lines 283-285 should be combined to one

File: api-client.php [hunk 2]
```diff
@@ -198,94 +241,123 @@ public function login($username = "", $password = "", $theme = "PaperLantern")
...
-        $databases = array();
-        $aDataTableDetailHTML = $this->getTableElements($this->cpanelUrl . "/panel/indexpl.php?option=pma");
+        $databases = [];
+        $aDataTableDetailHTML = $this->getTableElements(
+            $this->cpanelUrl . "/panel/indexpl.php?option=pma"
+        );
         foreach ($aDataTableDetailHTML as $database) {
-            $databases[str_replace($this->accountUsername . "_", "", array_shift($database))] = true;
+            $databases[
+                str_replace(
```
Comment (greenreader9) [inline api-client.php:300]: 300-303 can probably be combined for readability, no?

File: api-client.php [hunk 3]
```diff
@@ -484,13 +607,11 @@ public function showErrorPage($domainname = "", $option = "400")
         $htmlContent = $this->simpleCurl(
             $this->cpanelUrl . "/panel/indexpl.php?option=errorpages_configure",
             true,
-            array(
-                "domain_name" => $domainname
-            ),
+            [
```
Comment (greenreader9) [inline api-client.php:610]: Combine 610-612

File: api-client.php [hunk 4]
```diff
@@ -509,55 +630,53 @@ public function updateErrorPages(
...
         }
         $htmlContent = $this->simpleCurl(
-            $this->cpanelUrl . "/panel/indexpl.php?option=phpchangeconfig_configure",
+            $this->cpanelUrl .
+                "/panel/indexpl.php?option=phpchangeconfig_configure",
             true,
-            array(
-                "domain_name" => $domainname
-            ),
+            [
```
Comment (greenreader9) [inline api-client.php:675]: Combine 675-677

**Review by greenreader9 — CHANGES_REQUESTED** [review]
Readability, prevent 🙄 when viewing

**Comment by SpookyKipper** [comment]
These commits should do.

**Comment by SpookyKipper** [comment]
The above commit adds getCNAMErecords

**Comment by SpookyKipper** [comment]
Adds
`createCNAMErecord($source, $domain, $dest)`
`deleteCNAMErecord($source)`

**Comment by SpookyKipper** [comment]
Adds full controls of MX and SPF records

**Pull request merged**